### DISCUSSION TITLE
Ref order for conditional field

### DISCUSSION THREAD

**Discussion by specialdoom** [react-hook-form/react-hook-form#7523] [Q&A]
I have a form with 3 fields (F1, F2, F3) and 1 conditional field ( F2 it's dependent on an existing field and it renders only if that field has a value; it's position it's between the F1 and F3 fields ). All the fields has a required validation and if I click on submit before doing anything the first field it's being focused because of the required validation. But for some reason when I select the F1 and click on submit the focus it's on F3 and not F2. Is this an issue or it's the implicit behaviour?

Stackblitz: https://stackblitz.com/edit/react-mdjicw

Steps to reproduce:
1. Select value from the first select
2. Go to bottom of the page
3. Click submit
4. Focus it's on F3 field, and not F2

But for some reason when I use reset and reset the entire form everything it's alright.

One thing to notice it's that the order inside `errors` object, after selecting F1 and submit, it's 

```
{
  F3: ...,
  F2: ...
}
```

If this is the implicit behaviour, is it save to accomplish this using reset like this
```
reset({...getValues(), F2: defaultValue});
```

**Comment by Moshyfawn** [comment]
I believe, it's an API design limitation. There's no fully-fledged focus mngmnt for RHF atm. See issues like #6432 for some ctx.
One thing I can suggest is to use a validation schema w/ [@hookform/resolvers](https://github.com/react-hook-form/resolvers). That way your schema will dictate the order of the fields and should allow for the correct focus management on error.
See [this Stackblitz](https://stackblitz.com/edit/validation-schema-fields-order-example)
